actix-http/src/h1/dispatcher.rs

```diff
@@ -444,14 +449,9 @@ where
                 }
             };
 
-            this = self.as_mut().project();
-
-            // set new state
-            if let Some(state) = state {
-                this.state.set(state);
-                if !self.state.is_empty() {
-                    continue;
-                }
+            // state is changed and continue when the state is not Empty
+            if state_change && !self.state.is_empty() {
+                continue;
```

Comment:
the logic is different when state is changed and new state is empty

Reply:
as in, that case now falls through to the else branch instead of doing nothing

Reply:
`State::Empty` was also considered a state change and returned in Ok() part of `send_response`.
It would also goes in to the Some branch.

Reply:
Ok. Would fix it